stream-rpc/src/connection/manager.rs

```diff
@@ -0,0 +1,129 @@
+// Copyright (c) The Diem Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{
+    connection::{BoxConnectionStream, Client, ConnectionContext, StreamSender},
+    subscriptions::SubscriptionConfig,
+    counters, logging,
+};
+use diem_logger::debug;
+use futures::StreamExt;
+use std::{
+    collections::HashMap,
+    sync::{
+        atomic::{AtomicU64, Ordering},
+        Arc, RwLock,
+    },
+};
+use storage_interface::DbReader;
+
+/// Our global unique user id counter.
+static NEXT_USER_ID: AtomicU64 = AtomicU64::new(0);
```

Comment:
There's no reason for this to be a global, lets stick it inside of the ConnectionManager.